pyro/distributions/transforms/block_autoregressive.py

```diff
@@ -109,17 +112,22 @@ def _call(self, x):
         y = x
         for idx in range(len(self.layers)):
             pre_activation, dy_dx = self.layers[idx](y.unsqueeze(-1))
-            J_act = self.log_df_dx(pre_activation).view(*(list(x.size()) + [-1, 1]))
 
             if idx == 0:
+                y = self.T._call(pre_activation)
+                J_act = self.T.log_abs_det_jacobian((pre_activation).view(
+                    *(list(x.size()) + [-1, 1])), y.view(*(list(x.size()) + [-1, 1])))
```

Comment:
I think that it is better to avoid computing `y` and simply use `self.T.log_abs_det_jacobian(x, None)`. But the current version of `sigmoid` activation has logdet depending on `y` (it is fine with the master version of PyTorch), so it is up to you to decide what is the best solution. Maybe we can leave a TODO comment here to address it in the future.

Reply:
Computing `y` is necessary for the next iteration of the loop

Reply:
yup, you are right. :D